mlflow/utils/autologging_utils/__init__.py

```diff
@@ -376,11 +376,12 @@ def autolog(*args, **kwargs):
             except Exception:
                 pass
 
+            revert_patches(name)
```

Comment:
Before setup new patches, we should revert all existing patches first. different autolog config may setup different patches, if we don't revert existing patches first, it  will make things messy.

Reply:
Makes sense! Can we add a test case for something that would fail if we didn't have this fix implemented?